src/Scientist/Internals/ExperimentInstance.cs

```diff
@@ -106,12 +129,37 @@ internal ExperimentInstance(string name, NamedBehavior control, IEnumerable<Name
                 return false;
             }
 
-            //TODO: Does this really need to be async? We could run sync and return on first true
-            var results = await Task.WhenAll(Ignores.Select(i => i(control.Value, candidate.Value)));
+            try
+            {
+                //TODO: Does this really need to be async? We could run sync and return on first true
+                var results = await Task.WhenAll(Ignores.Select(i => i(control.Value, candidate.Value)));
 
-            return results.Any(i => i);
+                return results.Any(i => i);
+            }
+            catch (Exception ex)
+            {
+                Thrown(Operation.Ignore, ex);
+                return false;
+            }
         }
         
+        /// <summary>
+        /// Determine whether or not the experiment should run.
+        /// </summary>
+        async Task<bool> ShouldExperimentRun()
+        {
+            try
+            {
+                // TODO Implement Enabled here.
+                return Behaviors.Count > 1 && await RunIfAllows();
```

Comment:
The count should be greater than one? Is that because the behaviors collection includes both the control and the candidate? If so, might be useful to comment that.